### ISSUE TITLE
apache2_module always lists enabled modules with dependencies as 'changed'

### ISSUE THREAD

**Issue by HontoNoRoger** [ansible/ansible#7736]
In Ansible Version 1.6.2 the module 'apache2_module', which was added in 1.6, always lists apache modules when being enabled as 'changed' if they have some dependencies.

Tested with the modules headers, deflate, rewrite, ssl.
Here's the output from Ansible:

```
TASK: [apache | Enable standard mods] ***************************************** 
<ansibleCrawler> REMOTE_MODULE apache2_module name=headers state=present
ok: [ansibleCrawler] => (item=headers) => {"changed": false, "item": "headers", "result": "Success"}
<ansibleCrawler> REMOTE_MODULE apache2_module name=deflate state=present
changed: [ansibleCrawler] => (item=deflate) => {"changed": true, "item": "deflate", "result": "Enabled"}
<ansibleCrawler> REMOTE_MODULE apache2_module name=rewrite state=present
ok: [ansibleCrawler] => (item=rewrite) => {"changed": false, "item": "rewrite", "result": "Success"}
<ansibleCrawler> REMOTE_MODULE apache2_module name=ssl state=present
changed: [ansibleCrawler] => (item=ssl) => {"changed": true, "item": "ssl", "result": "Enabled"}
```

And here's what manual execution of a2enmod on the target machine prints out:

```
root@targetMachine ~ # a2enmod headers
Module headers already enabled
root@targetMachine ~ # a2enmod deflate
Considering dependency filter for deflate:
Module filter already enabled
Module deflate already enabled
root@targetMachine ~ # a2enmod rewrite
Module rewrite already enabled
root@targetMachine ~ # a2enmod ssl
Considering dependency setenvif for ssl:
Module setenvif already enabled
Considering dependency mime for ssl:
Module mime already enabled
Considering dependency socache_shmcb for ssl:
Module socache_shmcb already enabled
Module ssl already enabled
```

As you can see, this only involves modules which depend on others.
If using apache2_module with the notify directive, it would cause to always trigger the notification, e.g. restarting Apache.

**Comment by mjonuschat** [comment]
+1 for getting this merged

**Comment by boedah** [comment]
+1 from me also

**Comment by mjonuschat** [comment]
@mpdehaan @jimi-c: There's has been a fix in the develop branch for about 3 months (7883261335fb4d6a6a67f29ee539c5cc126f9590), any chance of getting that into the next release? Somehow it didn't make it into any of the later 1.6/1.7 releases

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the commit merged in via PR #7923 should resolve this problem for you. This will be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!
